src/librustc_codegen_llvm/builder.rs

```diff
@@ -1062,6 +1062,15 @@ impl BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
     fn do_not_inline(&mut self, llret: &'ll Value) {
         llvm::Attribute::NoInline.apply_callsite(llvm::AttributePlace::Function, llret);
     }
+
+    fn phi(&mut self, ty: &'ll Type, vals: &[&'ll Value], bbs: &[&'ll BasicBlock]) -> &'ll Value {
+        assert_eq!(vals.len(), bbs.len());
+        let phi = unsafe { llvm::LLVMBuildPhi(self.llbuilder, ty, UNNAMED) };
+        unsafe {
+            llvm::LLVMAddIncoming(phi, vals.as_ptr(), bbs.as_ptr(), vals.len() as c_uint);
+            phi
+        }
+    }
```

Comment:
Please just make the `phi` below `pub(crate)` instead. I removed `phi` from `BuilderMethods` as not every codegen backend uses phi nodes. Cranelift for example uses basic block parameters instead.